examples/advanced/pytorch_ddp_example/main.py

```diff
@@ -0,0 +1,42 @@
+import os
+
+import torch
+import torch.distributed as dist
+from omegaconf import DictConfig
+
+import hydra
+
+
+def setup(master_addr: str, master_port: str, rank: int, world_size: int, backend: str):
+    """Initializes distributed process group.
+
+    Arguments:
+        rank: the rank of the current process.
+        world_size: the total number of processes.
+        backend: the backend used for distributed processing.
+    """
+    os.environ["MASTER_ADDR"] = master_addr
+    os.environ["MASTER_PORT"] = master_port
+    dist.init_process_group(backend=backend, rank=rank, world_size=world_size)
+
+
+def cleanup():
+    """Cleans up distributed backend resources."""
+    dist.destroy_process_group()
+
+
+@hydra.main(config_path="conf", config_name="config")
+def main(cfg: DictConfig):
+    setup(cfg.master_addr, cfg.master_port, cfg.rank, cfg.world_size, cfg.backend)
+    group = dist.new_group(list(range(cfg.world_size)))
+    tensor = torch.rand(1)
```

Comment:
You can fill the tensors according to the rank and make sure the average is what you expect (not random).